File: calculator.rs
use std::io;

fn main() {
    // Read the first number
    let num1: f64 = get_user_input("Enter the first number:");

    // Read the operator
    let operator: char = get_user_input("Enter the operator (+, -, *, /):");

    // Read the second number
    let num2: f64 = get_user_input("Enter the second number:");

    // Perform the calculation based on the operator
    let result = match operator {
        '+' => num1 + num2,
        '-' => num1 - num2,
        '*' => num1 * num2,
        '/' => {
            if num2 != 0.0 {
                num1 / num2
            } else {
                println!("Error: Division by zero!");
                return;
            }
        }
        _ => {
            println!("Error: Invalid operator!");
            return;
        }
    };

    // Display the result
    println!("Result: {}", result);
}

// Function to get user input and parse it as f64
fn get_user_input(prompt: &str) -> f64 {
    loop {
        println!("{}", prompt);

        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Failed to read line");

        match input.trim().parse() {
            Ok(num) => return num,
            Err(_) => println!("Invalid input. Please enter a valid number."),
        }
    }
}

File: calculator.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	if len(os.Args) != 4 {
		fmt.Println("Usage: calculator <number> <operator> <number>")
		os.Exit(1)
	}

	// Parse command line arguments
	num1, err := strconv.ParseFloat(os.Args[1], 64)
	if err != nil {
		fmt.Println("Error parsing number 1:", err)
		os.Exit(1)
	}

	operator := os.Args[2]

	num2, err := strconv.ParseFloat(os.Args[3], 64)
	if err != nil {
		fmt.Println("Error parsing number 2:", err)
		os.Exit(1)
	}

	// Perform calculation based on the operator
	var result float64

	switch operator {
	case "+":
		result = num1 + num2
	case "-":
		result = num1 - num2
	case "*":
		result = num1 * num2
	case "/":
		if num2 == 0 {
			fmt.Println("Error: Division by zero")
			os.Exit(1)
		}
		result = num1 / num2
	default:
		fmt.Println("Invalid operator:", operator)
		os.Exit(1)
	}

	// Print the result
	fmt.Printf("Result: %v %s %v = %v\n", num1, operator, num2, result)
}
